src/Color/AlphaColor.ts

```diff
@@ -0,0 +1,68 @@
+import Color from './Color';
+import Utils from '../Utils';
+
+class AlphaColor extends Color {
+  alpha : number;
+
+  constructor(red: number, green: number, blue: number, alpha: number) {
+    super(red, green, blue);
+    this.alpha = alpha;
+  }
+
+  toRgba() : string {
+    return `rgba(${this.red},${this.green},${this.blue},${this.alpha / 255})`;
+  }
+
+  equal(color : AlphaColor) : boolean {
+    return color.red === this.red
+      && color.green === this.green
+      && color.blue === this.blue
+      && color.alpha === this.alpha;
+  }
+
+  hueLightEqual(color : AlphaColor) : boolean {
+    const hl1 = this.getHueLight();
+    const hl2 = color.getHueLight();
+    return hl1[0] === hl2[0] && hl1[1] === hl2[1];
+  }
+
+  getHueLight() : Array<number> {
+    const opacity = this.alpha / 255;
+
+    const red = ((1 - opacity) * 255 + opacity * this.red) / 255;
+    const green = ((1 - opacity) * 255 + opacity * this.green) / 255;
+    const blue = ((1 - opacity) * 255 + opacity * this.blue) / 255;
+
+    const max = Math.max(red, green, blue);
+    const min = Math.min(red, green, blue);
+
+    const light = (max + min) / 2;
+    let hue = light;
+
+    if (max === min) {
+      hue = 0;
+    } else {
+      const d = max - min;
+      // eslint-disable-next-line default-case
+      switch (max) {
+        case red: hue = (green - blue) / d + (green < blue ? 6 : 0); break;
+        case green: hue = (blue - red) / d + 2; break;
+        case blue: hue = (red - green) / d + 4; break;
+      }
+      hue /= 6;
+    }
+
+    return [Utils.round(hue), Utils.round(light)];
+  }
```

Comment:
Ce qui me gêne c'est ca en effet `return [Utils.round(hue), Utils.round(light)];`. Je créer une classe mais je laisse met une fonction static qui prend deux alphacolor pour pouvoir check l'égalité facilement.

Reply:
Now I do understand what you're telling me, here's what I propose, tell me if you feel like this would be better or not, and if it's not, you can resolve the conversation.

1. Create a HueLight class.
```ts
class HueLight {
  hue: number;
  light: number;
  // ...
}
```
2. Move `getHueLight` as the `constructor` of the HueLight class. Do not return the array as you used to do in getHueLight: instead, assign the properties of HueLight.
```ts
class HueLight {
  constructor(red: number, green: number, blue: number, opacity: number) {
    // Does whatever getHueLight used to do.
    // ...
    // Delete this line!
    return [Utils.round(hue), Utils.round(light)];
    // Instead do this:
    this.hue = Utils.round(hue);
    this.light = Utils.round(light);
  }
```
3. Create an easy and clear equal function in HueLight:
```ts
HueLight {
  // ...
  equal(e: HueLight) {
    return this.hue === e.hue && this.light === e.light;
  }
```
4. Create a `toHueLight` or `asHueLight` function in AlphaColor:
```ts
asHueLight(): HueLight {
  return new HueLight(this.red, this.green, this.blue, this.opacity);
}
```
The only place where hueLightEqual is used is in Fill.ts, so now this code:
```ts
if (this.baseColor.hueLightEqual(this.drawColor)) return;
```
would become:
```ts
if (this.baseColor.asHueLight().equal(this.drawColor.asHueLight())) return;
```

Reply:
Cf discussion de discord, j'ai fais un commit

Reply:
Edit : si je fais une classe toHueLight dans AlphaColor ca me fais une cycle dependencie. Je laisse donc comme c'est.